File: backend/auth-service/src/main/java/com/auth/app/services/application/AuthService.java
package com.auth.app.services.application;

import com.auth.app.domain.valueObjects.IpAddress;
import com.auth.app.domain.valueObjects.UserAgent;
import com.auth.app.dto.request.ChangePasswordRequest;
import com.auth.app.dto.request.ForgotPasswordRequest;
import com.auth.app.dto.request.LoginRequest;
import com.auth.app.dto.request.RegisterRequest;
import com.auth.app.dto.request.ResetPasswordRequest;
import com.auth.app.dto.response.AuthResponse;
import com.auth.app.dto.response.UserResponse;


public interface AuthService {
    AuthResponse register(RegisterRequest request, IpAddress ipAddress, UserAgent userAgent);
    void resendVerificationEmail(String email, IpAddress ipAddress, UserAgent userAgent);
    void verifyEmail(String rawToken, IpAddress ipAddress, UserAgent userAgent);
    AuthResponse login(LoginRequest request, IpAddress ipAddress, UserAgent userAgent);
    AuthResponse refresh(String rawRefreshToken, IpAddress ipAddress, UserAgent userAgent);
    void logout(String authorizationHeader, IpAddress ipAddress, UserAgent userAgent);
    void changePassword(String authorizationHeader, ChangePasswordRequest request, IpAddress ipAddress, UserAgent userAgent);
    void forgotPassword(ForgotPasswordRequest request, IpAddress ipAddress, UserAgent userAgent);
    void resetPassword(ResetPasswordRequest request, IpAddress ipAddress, UserAgent userAgent);
    UserResponse getCurrentUser(String authorizationHeader, IpAddress ipAddress, UserAgent userAgent);
}

File: frontend/services/authService.ts
import api from '@/lib/api';
import { ChangePasswordRequest } from '@/types/Request/ChangePasswordRequest';
import { LoginRequest } from '@/types/Request/LoginRequest';
import { RegisterRequest } from '@/types/Request/RegisterRequest';
import { ResetPasswordRequest } from '@/types/Request/ResetPasswordRequest';
import { ApiResponse } from '@/types/Response/ApiResponse';
import { AuthResponse } from '@/types/Response/AuthResponse';

const BASE_URL = "/auth";

export const authService = {
    login: (credentials: LoginRequest) => api.post<ApiResponse<AuthResponse>>(
        `${BASE_URL}/login`, credentials
    ),

    logout: () => api.post<ApiResponse<AuthResponse>>(
        `${BASE_URL}/logout`
    ),

    register: (credentials: RegisterRequest) => api.post<ApiResponse<AuthResponse>>(
        `${BASE_URL}/register`, credentials
    ),

    resendVerificationEmail: (email: string) => api.post<ApiResponse<AuthResponse>>(
        `${BASE_URL}/resend-verification`, { email }
    ),

    verifyEmail: (token: string) => api.post<ApiResponse<AuthResponse>>(
        `${BASE_URL}/verify`, null, { params: { token } }
    ),

    forgotPassword: (email: string) => api.post<ApiResponse<AuthResponse>>(
        `${BASE_URL}/forgot-password`, { email }
    ),

    resetPassword: (data: ResetPasswordRequest) => api.post<ApiResponse<AuthResponse>>(
        `${BASE_URL}/reset-password`, data
    ),

    changePassword: (data: ChangePasswordRequest) => api.post<ApiResponse<AuthResponse>>(
        `${BASE_URL}/change-password`, data
    ),

    me: () => api.get<ApiResponse<AuthResponse>>(
        `${BASE_URL}/me`
    ),

    refresh: () => api.post<ApiResponse<AuthResponse>>(
        `${BASE_URL}/refresh`
    ),
};
